pkg/servicemesh/ossmcommon/operations_unit_test.go

```diff
@@ -0,0 +1,72 @@
+package ossmcommon_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/servicemesh/ossmcommon"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var _ = Describe("SetGatewayName", func() {
+	var (
+		tempDir                             string
+		dscispec                            *dsci.DSCInitializationSpec
+		KfnotebookControllerServiceMeshPath string
+	)
+
+	BeforeEach(func() {
+		var err error
+
+		tempDir, err = os.MkdirTemp("", "ossm-test")
+		Expect(err).NotTo(HaveOccurred())
+
+		tempFilePath := filepath.Join(tempDir, "ossm.env")
+		tempFile, err := os.Create(tempFilePath)
+		Expect(err).NotTo(HaveOccurred())
+
+		mockContents := "ISTIO_GATEWAY=defaultnamespace/odh-gateway\nAnotherSetting=value"
+		_, err = tempFile.WriteString(mockContents)
+		Expect(err).NotTo(HaveOccurred())
+		tempFile.Close()
+
+		// Mock needed vars
+		KfnotebookControllerServiceMeshPath = tempDir
+		dscispec = &dsci.DSCInitializationSpec{
+			ApplicationsNamespace: "testnamespace",
+		}
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tempDir)
+	})
+
+	It("should replace gateway name in the file", func() {
+		err := ossmcommon.SetGatewayName(dscispec, KfnotebookControllerServiceMeshPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		updatedContents, err := os.ReadFile(filepath.Join(KfnotebookControllerServiceMeshPath, "ossm.env"))
+		Expect(err).NotTo(HaveOccurred())
+
+		expected := "ISTIO_GATEWAY=testnamespace/odh-gateway"
+		Expect(strings.Contains(string(updatedContents), expected)).To(BeTrue(), "Expected content to contain %q, got %q", expected, updatedContents)
```

Comment:
I think we can leverage built-in Gomega matchers for simplicity https://onsi.github.io/gomega/#working-with-strings-json-and-yaml